x-pack/plugins/stack_alerts/public/alert_types/threshold/expression.tsx

```diff
@@ -261,6 +288,33 @@ export const IndexThresholdAlertTypeExpression: React.FunctionComponent<
         }
       />
       <EuiSpacer />
+      <EuiTitle size="xs">
+        <h5>
+          <FormattedMessage
+            id="xpack.stackAlerts.threshold.ui.filterTitle"
+            defaultMessage="Filter (Optional)"
+          />
+        </h5>
+      </EuiTitle>
+      <EuiSpacer size="s" />
+      <EuiFormRow
+        helpText={i18n.translate('xpack.stackAlerts.threshold.ui.filterKQLHelpText', {
+          defaultMessage: 'Use a KQL expression to limit the scope of your alert trigger.',
+        })}
+        fullWidth
+        display="rowCompressed"
+        isInvalid={errors.filterKuery.length > 0}
+        error={errors.filterKuery}
+      >
+        <EuiFieldSearch
+          data-test-subj="filterKuery"
+          onChange={handleFilterChange}
```

Comment:
Can we add a debounce here so the validation is not called with each keypress?

Reply:
This is a managed form handleFilterChange sets the filed value as well.
i'll try to separate validation.